packages/agent/src/utils/callAndPoll.ts

```diff
@@ -0,0 +1,54 @@
+import { Principal } from '@dfinity/principal';
+import { Agent, Certificate, bufFromBufLike, polling, v3ResponseBody } from '..';
+import { AgentError } from '../errors';
+
+/**
+ * Call a canister using the v3 api and either return the response or fall back to polling
+ * @param options - The options to use when calling the canister
+ * @param options.canisterId - The canister id to call
+ * @param options.methodName - The method name to call
+ * @param options.agent - The agent to use to make the call
+ * @param options.arg - The argument to pass to the canister
+ * @returns The certificate response from the canister (which includes the reply)
+ */
+export async function callAndPoll(options: {
+  canisterId: Principal | string;
+  methodName: string;
+  agent: Agent;
+  arg: ArrayBuffer;
+}): Promise<ArrayBuffer> {
+  const { canisterId, methodName, agent, arg } = options;
+  const cid = Principal.from(options.canisterId);
+
+  const { defaultStrategy } = polling.strategy;
+
+  if (agent.rootKey == null) throw new Error('Agent root key not initialized before making call');
+
+  const { requestId, response } = await agent.call(cid, {
+    methodName,
+    arg,
+    effectiveCanisterId: cid,
+  });
+
+  let certificate: Certificate;
+  if (response.body && (response.body as v3ResponseBody).certificate) {
```

Comment:
Should you assert that the type is v3 before casting? This feels a bit optimistic.